controller_manager:
  ros__parameters:
    update_rate: 500 

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    arm_controller:
      type: position_controllers/JointGroupPositionController

joint_state_broadcaster:
  ros__parameters:
    publish_rate: 50

arm_controller:
  ros__parameters:
    joints:
      - link_1_joint
      - link_2_joint
      - link_3_joint
      - forearm_joint
      - differential_joint
      - gripper_joint
      - finger_1_joint
      - finger_2_joint      
    publish_rate: 50
    gains:
      link_1_joint:      {p: 100.0, i: 0.01, d: 10.0}
      link_2_joint:      {p: 100.0, i: 0.01, d: 10.0}
      link_3_joint:      {p: 100.0, i: 0.01, d: 10.0}
      forearm_joint:      {p: 100.0, i: 0.01, d: 10.0}
      differential_joint: {p: 100.0, i: 0.01, d: 10.0}
      gripper_joint:      {p: 100.0, i: 0.01, d: 10.0}
      finger_1_joint:     {p: 100.0, i: 0.01, d: 10.0}
      finger_2_joint:     {p: 100.0, i: 0.01, d: 10.0}